### PR TITLE
Update the CI Toolkit and Xcode image

### PR DESCRIPTION
**Pull request by spencertransier**
This PR updates the Buildkite CI Xcode image to `14.3.1` and the CI Toolkit to `2.17.0`. 

This should be good to merge if CI is green! 

---

- [X] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

**Comment by mokagio** [comment]
Thanks @spencertransier!

I haven't investigated in this repo, but I think the CocoaPods issue we're seeing here is the same incompatibility with Xcode 14.3 seen elsewhere. E.g. https://github.com/wordpress-mobile/WordPress-iOS/pull/20801

I'd try updating CP to the latest version and trying again.

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ _None._
 
 ### Internal Changes
 
-_None._
+- Update the CI Xcode image to `14.3.1` and CI Toolkit to `2.17.0` [#258]
```
Comment (mokagio) [inline CHANGELOG.md]: FWIW, I don't see the need to track this information in the CHANGELOG. Granted, it is an internal change, but it does not affect the library's behavior nor requires developers to be mindful something different in the code.

**Review by mokagio — APPROVED** [review]
Thank you for keeping our stack up to date @spencertransier 🙌 

I pushed `8fb7433` which should address the podspec validation failure. It works on my machine™

<details>
<summary>Details</summary>

```
➜ be pod lib lint Automattic-Tracks-iOS.podspec

 -> Automattic-Tracks-iOS (2.2.0)
    - NOTE  | url: The URL (https://twitter.com/automattic) is not reachable.
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | xcodebuild:  note: Building targets in dependency order
    - NOTE  | xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'SentryPrivate' from project 'Pods')
    - NOTE  | xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'Sentry' from project 'Pods')
    - NOTE  | xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'Sodium' from project 'Pods')
    - NOTE  | xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'Automattic-Tracks-iOS' from project 'Pods')
    - NOTE  | xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'App' from project 'App')
    - NOTE  | [iOS] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'UIDeviceIdentifier' from project 'Pods')
    - NOTE  | xcodebuild:  note: Using codesigning identity override:
    - NOTE  | [OSX] xcodebuild:  Pods.xcodeproj: warning: The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.11, but the range of supported deployment target versions is 10.13 to 13.3.99. (in target 'Sodium' from project 'Pods')
 ```

</details>

File: CHANGELOG.md (hunk 1 above)
Comment (spencertransier) [reply CHANGELOG.md]: Thanks for the feedback! I went ahead and removed it.

**Comment by spencertransier** [comment]
@mokagio 

> I pushed 8fb7433 which should address the podspec validation failure. It works on my machine™

Thank you for taking a look at this! I was researching how to resolve the validation failure and was thinking the only way to fix it was to have the minimum version of the UIDeviceIdentifier library bumped up to our minimum. 

For my learning, how did you track it down to that part of the code, and why does that code change fix the validation?

**Pull request merged**